docs/user-guides/retrieval.md

```diff
@@ -0,0 +1,282 @@
+╭───────────────────────────── Documents Summary ─────────────────────────────╮
+│                                                                             │
+│   Length                 1                                                  │
+│   Homogenous Documents   True                                               │
+│   Common Attributes      ('id', 'mime_type', 'text', 'tags', 'embedding')   │
+│   Multimodal dataclass   False                                              │
+│                                                                             │
+╰─────────────────────────────────────────────────────────────────────────────╯
+╭────────────────────── Attributes Summary ───────────────────────╮
+│                                                                 │
+│   Attribute   Data type      #Unique values   Has empty value   │
+│  ─────────────────────────────────────────────────────────────  │
+│   embedding   ('ndarray',)   1                False             │
+│   id          ('str',)       1                False             │
+│   mime_type   ('str',)       1                False             │
+│   tags        ('dict',)      1                False             │
+│   text        ('str',)       1                False             │
+│                                                                 │
+╰─────────────────────────────────────────────────────────────────╯
+```
+
+You don't need to call `client.encode()` explicitly since `client.index()` will handle this for you.
+
+
+Similar to `clip_server`, you can directly use the YAML config we have prepared for you by simple running:
+
+```bash
+python -m clip_server search_flow.yaml
+```
+
+The YAML config looks like this:
+```yaml
+jtype: Flow
+version: '1'
+with:
+  port: 51000
+executors:
+  - name: encoder
+    uses:
+      jtype: CLIPEncoder
+      metas:
+        py_modules:
+          - clip_server.executors.clip_torch
+  - name: indexer
+    uses:
+      jtype: AnnLiteIndexer
+      with:
+        dim: 512
+      metas:
+        py_modules:
+          - annlite.executor
+```
+
+| Parameter               | Description                                                                                                                  |
+|-------------------------|------------------------------------------------------------------------------------------------------------------------------|
+| `dim`                  | Dimension of embeddings. Default is 512 since the output dimension of `CLIP` model is 512.|
+
+
+## How to lower memory footprint?
+Sometimes the indexer will use a lot of memory because the HNSW indexer (which is used by `AnnLite`) is stored in memory. The efficient way to reduce memory footprint is dimension reduction. Retrieval in CLIP-as-service use [`Principal component analysis(PCA)`](https://en.wikipedia.org/wiki/Principal_component_analysis#:~:text=Principal%20component%20analysis%20(PCA)%20is,components%20and%20ignoring%20the%20rest.) to achieve this.
```

Comment:
Do we know _when_ it will use a lot of memory? Are there any drawbacks to PCA?
Otherwise this seems like a freebie that should always be done -> less memory, same performance

Reply:
Good idea,  will do it.